nearcore/src/runtime/mod.rs

```diff
@@ -704,16 +705,27 @@ impl RuntimeAdapter for NightshadeRuntime {
         self.flat_state_factory.get_flat_storage_state_for_shard(shard_id)
     }
 
-    #[cfg(feature = "protocol_feature_flat_state")]
     fn create_flat_storage_state_for_shard(
```

Comment:
Maybe rename this to "try_create_flat_storage_state_for_shard`, because the calling of this method doesn't guarantee the creation of flat storage state.